File: src/main/java/com/example/ecommerce/service/ProductService.java
package com.example.ecommerce.service;

import com.example.ecommerce.dto.ListProductDTO;
import com.example.ecommerce.dto.ProductDTO;
import com.example.ecommerce.dto.ProductPostDTO;
import com.example.ecommerce.dto.ResponseProductDTO;
import com.example.ecommerce.entity.Product;

import java.util.List;


public interface ProductService {
    ResponseProductDTO createProduct(ProductDTO productDTO);
    ProductDTO findById(Long id);
    List<Product> findAll(int pageNumber, int pageSize);
    ListProductDTO findByConditions(int pageNumber, int pageSize, String productName, String categoryName);
    ListProductDTO findByCategoryId(int pageNumber, int pageSize, Long categoryId);
    ResponseProductDTO updatedProductById(ProductDTO productDTO, Long productId);
    void deleteProductById(Long id);
}

File: frontend-reactjs/services/product.service.js
import axios from "axios"

const API_URL = "http://localhost:8080/api/v1/product"

class ProductService {

    getProductByCategoryId(categoryId, pageNumber, pageSize) {
        return axios
        .get(API_URL+`/?categoryId=${categoryId}`,{
            params: {
              pageNumber: pageNumber,
              pageSize: pageSize
            }
          })
    }

    getDetaiProduct(productId) {
      return axios
        .get(API_URL+`/${productId}`)
    }

    searchProduct(productName, pageNumber, pageSize) {
      return axios
        .get(API_URL+`/search`,{
          params: {
            productName : productName,
            pageNumber: pageNumber,
            pageSize: pageSize
          }
        })
    }
}

export default new ProductService();